packages/formspree-react/src/useForm.ts

```diff
@@ -5,50 +5,51 @@ import { useElements, CardElement, useStripe } from '@stripe/react-stripe-js';
 import { useFormspree } from './context';
 import type { ExtraData } from './types';
 import { version } from '../package.json';
-import type {
-  Client,
-  SubmissionResponse,
-  SubmissionData,
-  ErrorBody,
-  FormError,
+import {
+  appendExtraData,
+  type Client,
+  type FieldValues,
+  type SubmissionData,
+  type SubmissionSuccessResult,
+  type SubmissionErrorResult,
 } from '@formspree/core';
 
 type FormEvent = React.FormEvent<HTMLFormElement>;
 
-type SubmitHandler = (
-  submissionData: FormEvent | SubmissionData
-) => Promise<SubmissionResponse>;
+type SubmitHandler<T extends FieldValues> = (
+  submissionData: FormEvent | SubmissionData<T>
+) => Promise<void>;
```

Comment:
My thought for changing this to `Promise<void>` is that we design this to be a submit handler i.e. a function that triggers the submission that changes the state. Exposing the resolved value (result) or allowing it to reject with validation errors makes a confusing API IMO (when do I want to await on the hook vs reacting to the `formState`).

Given that we'll be adding a new hook (`useSubmit`) in a follow-up PR, I think `useForm` should focus on managing react state based on the submission result, when users want to manage the submission result/state themselves (or having another library that they use to manage state like react-hook-form) then they can use `useSubmit`.

I even thought about making this return `void` rather than `Promise<void>` to signal that this is a trigger function and nothing of interest is returned here.

Reply:
> I even thought about making this return void rather than Promise<void> to signal that this is a trigger function and nothing of interest is returned here.

Yeah I guess what is the advantage of Promise<void> over just void?

Reply:
My guess would be that it was not intentional, actually. I think it was just because the `handleSubmit` function was implemented as an async function so the returned type has to be `Promise<...>`. To make it return `void`, we just need to change it to a non-async function and either use `.then` instead of `await` or invoke an IIFE of an async function inside `handleSubmit` body.

> Returning just void makes the API even stricter. ... With just void, you really must rely on the react state hook, since that's the only information you have.

Agreed. Returning `Promise<void>` communicates that awaiting on the returned result of the handler has some meaning; it unintentionally expose implementation detail i.e. "when" the promise is resolved.

Changing the return type from `Promise<void>` to `void` is a breaking change though, and it should be a major version bump for `@formspree/react`.

Reply:
> Changing the return type from Promise<void> to void is a breaking change though, and it should be a major version bump for @formspree/react.

ah. Good point, but do you think we can get away with changing the resolved object in that case? Isn't that also a breaking change?